### ISSUE TITLE
Error in running my DEM model

### ISSUE THREAD

**Issue by sauravVerma92** [KratosMultiphysics/Kratos#6055]
![Capture5](https://user-images.githubusercontent.com/58203729/70379955-f3774b80-196e-11ea-8180-1ce840391ac0.JPG)
![Capture6](https://user-images.githubusercontent.com/58203729/70379956-f40fe200-196e-11ea-9bcf-c59413d2b317.JPG)

I am getting errors while running my program. I want to simulate the flow of particles from a barrel shaped pipe. So, I import the IGES file in the GiD, create separate layers and group named: wall, inlet, body. Then generate the mesh and create the MDPA and JSON file by clicking Calculate. I get the error both while running from the GiD and also when running from the Terminal.

DEM: Importing DEMApplication... ( compiled in mode "Release" ) done.
DEM: Running under OpenMP........ 
ModelPartIO:   [Reading Nodes    : 15034 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Elements : 15034 elements read] [Type: SphericParticle3D]
ModelPartIO:   [Total Lines Read : 75220]
ModelPartIO:   [Reading Nodes    : 5404 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Conditions : Traceback (most recent call last):
  File "KratosDEMAnalysis.py", line 34, in <module>
    DEMAnalysisStageWithFlush(model, project_parameters).Run()
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 49, in Run
    self.Initialize()
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/DEM_analysis_stage.py", line 265, in Initialize
    self.ReadModelParts()
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/DEM_analysis_stage.py", line 444, in ReadModelParts
    self.ReadModelPartsFromMdpaFile(max_node_id, max_elem_id, max_cond_id)
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/DEM_analysis_stage.py", line 420, in ReadModelPartsFromMdpaFile
    ReadModelPart(self.rigid_face_model_part, self.GetDEMWallsInputFileTag(), max_node_id, max_elem_id, max_cond_id)
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/DEM_analysis_stage.py", line 414, in ReadModelPart
    model_part_io.ReadModelPart(model_part)
RuntimeError: Error: Node #20440 is not found. [Line 5433 ]

in kratos/sources/model_part_io.cpp:4445:typename TContainerType::iterator ModelPartIO::FindKey(TContainerType &, TKeyType, string) [TContainerType = PointerVectorSet<Node,...>, TKeyType = unsigned long]
   kratos/sources/model_part_io.cpp:1552:void ModelPartIO::ReadConditionsBlock(ModelPartIO::NodesContainerType &, ModelPartIO::PropertiesContainerType &, ModelPartIO::ConditionsContainerType &)
   kratos/sources/model_part_io.cpp:511:virtual void ModelPartIO::ReadModelPart(ModelPart &)

**Comment by maceligueta** [comment]
Hi @sauravVerma92 . Apparently there is an inconsistency in the walls definition. It seems like in line 5433  of the file with ending 'DEM_FEM_Boundary.mdpa' a condition is writen which uses nodes that have not been listed before that point. It requires node 20440, but that one is not present in the first list of the file, where you provide the coordinates.
I do not know the exact reason for this, please check in GiD the mesh entities that are inside the Walls group, maybe it contains something that should not be there. 
Ultimately, you might upload the full example and we can look.

**Comment by sauravVerma92** [comment]
Thank u @maceligueta , I have found something very strange in the mdpa file. I will try to generate these files again in the GiD and let you know. Here are some lines from 5418 to 5436 of the DEM_FEM_Boundary.mdpa. Isn't the highlighted number are strange.

18229        2.23929       -0.00000       10.67108
18230        2.18315        0.49829       10.67108
18231        2.22521        0.25072       10.67108
End Nodes


Begin Nodes
End Nodes


Begin Nodes
End Nodes


Begin Conditions RigidFace3D3N // GUI DEM-FEM-Wall group identifier: wall
    1          3      17344 **-1881355059**          0 
    2          3      17198 **-1645006028**          0 
    3          3      17370 **-267101974**          0 
    4          3      17270 **-1064933786**          0

**Comment by sauravVerma92** [comment]
Hi @miguelmaso, I tried to regenerate the mdpa files from the GiD, but again the strange lines in the Conditions RigidFace3D3N section were generated. This time I deleted all the lines where negative values for the RigidFace3D3N Conditions were generated. Then the simulation runs but the results are printed only for 5 elements and it stops with an AssertionError. I don't understand why. Is there some other way to generate the mdpa files with my desirable initial conditions and boundary conditions. I have attached the gid folder here. It also contains the STEP file of the 3Dmodel that I am using.
[powderFlow6.gid.zip](https://github.com/KratosMultiphysics/Kratos/files/3948726/powderFlow6.gid.zip)

DEM: Importing DEMApplication... ( compiled in mode "Release" ) done.
DEM: Running under OpenMP........ 
ModelPartIO:   [Reading Nodes    : 15034 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Elements : 15034 elements read] [Type: SphericParticle3D]
ModelPartIO:   [Total Lines Read : 75220]
ModelPartIO:   [Reading Nodes    : 5404 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Nodes    : 0 nodes read]
ModelPartIO:   [Reading Conditions : 6080 conditions read] [Type: RigidFace3D3N]
ModelPartIO:   [Total Lines Read : 16926]
ModelPartIO:   [Total Lines Read : 1]
ModelPartIO:   [Reading Nodes    : 207 nodes read]
ModelPartIO:   [Total Lines Read : 462]
Initializing Problem...: 
DEM: Assigning DEM_D_Hertz_viscous_Coulomb to Properties 1
DEM: Assigning DEM_D_Hertz_viscous_Coulomb to Properties 2
-------------------------------------------------: 
  WARNING: Property WALL_FRICTION is deprecated: 
  since April 11th, 2018, replace with FRICTION: 
  Automatic replacement is done now.: 
-------------------------------------------------: 
Initialization Complete: 
DOFs for the DEM solution added correctly: 
DOFs for the DEM solution added correctly: 
DOFs for the DEM solution added correctly: 
DEM: ------------------DISCONTINUUM SOLVER STRATEGY---------------------

DEM:           **************************************************
DEM:             Parallelism Info:  MPI number of nodes: 1
DEM:             Parallelism Info:  OMP number of processors: 1
DEM:           **************************************************
DEM: 
Analysis: Analysis -START-  
Main loop starting...
DEM: Total number of TIME STEPs expected in the calculation: 120000
DEM: : 
*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (5 elements): 
                        (5 nodes): 

*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (5 elements): 
                        (5 nodes): 

Real time calculation: 4.000542879104614 seconds
DEM: In minutes: 0.06667571465174357 minutes
DEM: In hours: 0.0011112619108623928 hours
DEM: Simulation time: 0.09059999999999818 seconds
DEM: Percentage Completed:  1.51000 %
DEM: Computed time steps: 1812 out of 120000
DEM: : 
*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (4 elements): 
                        (4 nodes): 

*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (4 elements): 
                        (4 nodes): 

*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (4 elements): 
                        (4 nodes): 

Real time calculation: 8.001490831375122 seconds
DEM: In minutes: 0.1333581805229187 minutes
DEM: In hours: 0.002222636342048645 hours
DEM: Simulation time: 0.2040499999999857 seconds
DEM: Percentage Completed:  3.40083 %
DEM: Computed time steps: 4081 out of 120000
DEM: : 
*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (2 elements): 
                        (2 nodes): 

*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (2 elements): 
                        (2 nodes): 

*******************  PRINTING RESULTS FOR GID  ***************************: 
                        (0 elements): 
                        (0 nodes): 

Traceback (most recent call last):
  File "KratosDEMAnalysis.py", line 34, in <module>
    DEMAnalysisStageWithFlush(model, project_parameters).Run()
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 50, in Run
    self.RunSolutionLoop()
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 70, in RunSolutionLoop
    self.OutputSolutionStep()
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/DEM_analysis_stage.py", line 539, in OutputSolutionStep
    self.PrintResults()
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/DEM_analysis_stage.py", line 466, in PrintResults
    self.PrintResultsForGid(self.time)
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/DEM_analysis_stage.py", line 653, in PrintResultsForGid
    self.vtk_output.WriteResults(self.time)
  File "/Users/sauravverma/Kratos/bin/Release/KratosMultiphysics/DEMApplication/dem_vtk_output.py", line 107, in WriteResults
    hl.pointsToVTK(path, self.particles_X, self.particles_Y, self.particles_Z, {'radius':self.particles_R, 'material':self.particles_material, 'velocity':(self.velocities_X, self.velocities_Y, self.velocities_Z)})
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyevtk/hl.py", line 223, in pointsToVTK
    w.openPiece(ncells = npoints, npoints = npoints)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyevtk/vtk.py", line 252, in openPiece
    assert(False)
AssertionError

**Comment by maceligueta** [comment]
@sauravVerma92  Several files are missing in the .gid folder. Thus, I can't open it as a GiD project. I would like to do that, so I can check what is the exact problem (whether is a user mistake or a bug in the GUI).
Can you send it again, checking that it can be opened with GiD.?

**Comment by maceligueta** [comment]
@sauravVerma92  , I saw your issue very late, because we don't check all of them, just those where we are referenced. I recommend you to add @maceligueta or even better @KratosMultiphysics/dem  (the DEM Team) to get a fast response. If you tag us, we get an email notification. You can also `assign `the issue to someone.

**Comment by sauravVerma92** [comment]
Ok @maceligueta , I'll add people or the KratosMultiphysics/dem from next time.

Here, is the GiD file and the STEP file is inside the folder.

[powderFlow6.gid.zip](https://github.com/KratosMultiphysics/Kratos/files/3950103/powderFlow6.gid.zip)

**Comment by maceligueta** [comment]
Hello @sauravVerma92 . I just fixed the issue. Let's see if these steps help you out:

1. Update the GiDInterface Project, so you have the latest version of the GUI
2. Open your case
3. Remove the DEM-FEM Wall Condition you applied to the 'Walls' group
4. Add it again, making sure you select Surfaces
5. Mesh
6. Run

Apparently there was a bug which included spheres as part of the walls. I will keep track of this bug and see when it was fixed. @farrufat-cimne  just told me that he had seen this bug happening sometimes, but it was not fixed in his opinion.

**Comment by sauravVerma92** [comment]
@maceligueta I get the same error again.
![error](https://user-images.githubusercontent.com/58203729/70625813-2e72cb00-1c5e-11ea-8e39-3eba4d5d9957.JPG)

**Comment by sauravVerma92** [comment]
@maceligueta I also get this error this time.
![error1](https://user-images.githubusercontent.com/58203729/70640726-5d496b00-1c77-11ea-95a4-c132b048f2ff.JPG)

**Comment by sauravVerma92** [comment]
Hi @maceligueta ,  @KratosMultiphysics/dem @farrufat-cimne , The GiD doesn't seem to be compatible with MacOS Catalina, so I am facing problem using the interface. Can this be solved? Although, presently I am using the GiD in someone else Windows PC to generate the input files.

**Comment by maceligueta** [comment]
Thinking about what differences we might have, I thought that the GiD version could be different.
I am using GiD 14.1.7d (developer version, we always try to use the latest developer version). Could you please try if the updated interface works well with the latest GiD? It works well on my Linux. I am going to try it on a Mac.

**Comment by maceligueta** [comment]
We just tried on a MacOS Catalina and it worked! So the GiD version might be it.

**Comment by sauravVerma92** [comment]
Hi @maceligueta @ipouplana , 
When I try to add conditions for the wall in the GiD, I cannot scroll down to access the lower options. you can see it in the screenshot that I send. I am using the GiD14.1.7d in my 13'' MacOS Catalina.

<img width="1440" alt="Screenshot 2019-12-12 at 8 18 58 PM" src="https://user-images.githubusercontent.com/58203729/70718935-27b28980-1d2c-11ea-89a0-1af6309a0a07.png">

**Comment by maceligueta** [comment]
It seems like you are using an old version of GiDInterface, or that you are not on an updated 'master' branch. This mini window which you can't scroll was deactivated some time ago, precisely to avoid this kind of problems. When I open your case I get a message like 'problem is going to be updated' or something similar. After that, I found it totally impossible to reach that mini-window you are signaling. Please @jginternational  correct me if I am wrong.

**Comment by maceligueta** [comment]
(sorry @sauravVerma92 , @jginternational  is our expert on GiD GUI, and he coded some parts of the DEM GUI too, that's why I am including him in the loop)

**Comment by jginternational** [comment]
Right, the current interface looks like this:
![image](https://user-images.githubusercontent.com/5918085/70722795-3121fe80-1cf8-11ea-8189-d6f5573683b0.png)

**Comment by sauravVerma92** [comment]
Hi @miguelmaso @jginternational ,
Yes I was not using the latest version of GiDInterface. I was simply using the preinstalled kratos from the problemtype. So I deleted the previous kratos.gid,  then downloaded the GiDInterface and in the GiD GUI I go to Problemtype>load>the location where GiDInterface_Master is donloaded>kratos.gid. 
Next I navigate to kratos.gid/exec/  in the terminal
and type this in the terminal :  ln -s ~/Kratos Kratos
after that I copy this kratos.gid and paste it in the Applications/GiD-14.1.7d.app/Contents/MacOS/problemtypes/

Now I again open the GiD GUI and load kratos from the problemtype and choose 3D DEM. Then I imported the IGES file and gave all the necessary conditions. Mesh and Calculate. Then I get this error:

/Applications/GiD-14.1.7d.app/Contents/MacOS/problemtypes/kratos.gid/kratos.unix.bat: line 48: /Applications/GiD-14.1.7d.app/Contents/MacOS/problemtypes/kratos.gid/exec/Kratos/runkratos: No such file or directory

**Comment by maceligueta** [comment]
Thanks @sauravVerma92 . Look for the file called 'runkratos'. It is probably not located in ~/Kratos.
Maybe in ~/Kratos/bin/Release/ ?
If so, you should have typed: 
`ln -s ~/Kratos/bin/Release Kratos `
instead of what you did. We are still polishing that help file. Just make sure that the link you are creating points to the right location.

**Comment by sauravVerma92** [comment]
Thanks @maceligueta , I tried it but still got the same error in the GiD GUI. 
But then I copied the KratosDEMAnalysis.py file from the examples and paste it in the main powderFlow7.gid folder and run it from Terminal. The program is running now!! and the output files are also being generated.
So it means that the problem of incorrect input files is now solved. I can generate the proper input files now. Thank you.
In the GiD I think I did some mistake in creating the link to the kratos.gid
